dot/network/host_test.go

```diff
@@ -273,24 +271,32 @@ func TestExistingStream(t *testing.T) {
 	require.NoError(t, err)
 
 	time.Sleep(TestMessageTimeout)
```

Comment:
can we refactor this test to not use sleeps anymore? Now that you have `testStreamHandler`, you can write messages into an internal channel from `testStreamHandler.handleMessage` and just listen on that channel.